File: app/src/main/java/com/emarsys/catseeker/network/SelinaService.java
package com.emarsys.catseeker.network;

import com.emarsys.catseeker.model.CatMessageModel;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;

interface SelinaService {

    @POST("services/{secret}")
    Call<Void> postCat(
            @Path(value = "secret", encoded = true) String secret,
            @Body CatMessageModel catMessageModel);
}

File: app/src/main/kotlin/com/emarsys/catseeker/network/SelinaService.kt
package com.emarsys.catseeker.network

import com.emarsys.catseeker.model.CatMessageModel
import com.emarsys.catseeker.model.MetaData
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Path

interface SelinaService {
    @POST("services/{secret}")
    fun postCat(
            @Body catMessageModel: CatMessageModel,
            @Path(value = "secret", encoded = true) secret: String = MetaData.apiSecret
    ): Call<Void>
}